### ISSUE TITLE
Regression in 1.9 ?

### ISSUE THREAD

**Issue by benrom** [c-rack/quantum-elixir#160]
Since 1.9 my jobs are not executed any more.

I tested the simple:

```
      job = %Quantum.Job{schedule: "* * * * *", task: fn -> Logger.info("tick") end}
      Quantum.add_job(:ticker, job)
```

within my app start function (umbrella project, but Quantum within only one sub app).
That code works with 1.8.1 (I can see the log entries every minute) but not with the brand new 1.9 (no other change done to code/config).

`Quantum.add_job` returns ok in both cases. If I run `Quantum.jobs` from the console I can see my job listed. Tested with both elixir 1.3 and 1.4.

Anybody faces the same problem?

**Comment by maennchen** [comment]
I've setup a test quantum app. I can reproduce your bug. It seems that only jobs from the config are executed, but not the ones defined in runtime.

Minimal Example: https://github.com/jshmrtn/quantum-test

**Comment by maennchen** [comment]
The problem is that the normalizer is not normalizing values of a `%Quantum.Job{}` struct.

I'll create a PR in a few minutes.

**Comment by maennchen** [comment]
@benrom Could you check the proposed solution? #161

**Comment by benrom** [comment]
I've just tested with master, It does work! Thanks a lot @maennchen.
Waiting for the new hex release.

**Comment by benrom** [comment]
Actually I have a problem in production:

```
(UndefinedFunctionError) function Crontab.CronExpression.Parser.parse!/1 is undefined (module Crontab.CronExpression.Parser is not available)             Crontab.CronExpression.Parser.parse!("*/15 * * * *")
```

Is there a dependency problem here?

**Comment by maennchen** [comment]
@benrom Yes, `v1.9.0` depends on [crontab](https://hex.pm/packages/crontab) for the cron expressions. You'll need `:crontab` in you applications.

@c-rack Is the change that adds `crontab` to the application tree already deployed to hex?

**Comment by benrom** [comment]
I did try to add :crontab to my application, it didn't fix it. crontab is a dependency of quantum as defined in its mix.exs. Perhaps it's because I was using the github (master) mix dependency instead of the hex one?

**Comment by maennchen** [comment]
@benrom I don't think this should be a problem.

I have `v1.9.0` in my application which uses `distillery` as well. I don't have those problems.

Could you try to add `:crontab` into your release applications and see if this solves the issue? (This isn't a solution, but helps us identifying the issue.)

**Comment by benrom** [comment]
If I use v1.9.0, will that parsing code be run (considering the issue above was not fixed yet), thus will I see the error when running it in production?

**Comment by maennchen** [comment]
Yes, `v1.9.0` already uses the crontab library. You'll see that error in production too.

It'll although only be parsed if you either add the job in the configuration or use the `Crontab.CronExpression` sigil, since the newest master is not released yet. (The cause of the issue that dynamic jobs are not executed, was that they were not parsed.)
